kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/serialization/AxonSerializers.kt

```diff
@@ -0,0 +1,333 @@
+package org.axonframework.extensions.kotlin.serialization
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.PolymorphicSerializer
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.builtins.MapSerializer
+import kotlinx.serialization.builtins.SetSerializer
+import kotlinx.serialization.builtins.nullable
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.descriptors.element
+import kotlinx.serialization.encoding.CompositeDecoder
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.encoding.decodeStructure
+import kotlinx.serialization.encoding.encodeStructure
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.modules.polymorphic
+import kotlinx.serialization.modules.subclass
+import org.axonframework.eventhandling.GapAwareTrackingToken
+import org.axonframework.eventhandling.GlobalSequenceTrackingToken
+import org.axonframework.eventhandling.MergedTrackingToken
+import org.axonframework.eventhandling.MultiSourceTrackingToken
+import org.axonframework.eventhandling.ReplayToken
+import org.axonframework.eventhandling.TrackingToken
+import org.axonframework.eventhandling.scheduling.ScheduleToken
+import org.axonframework.eventhandling.scheduling.java.SimpleScheduleToken
+import org.axonframework.eventhandling.scheduling.quartz.QuartzScheduleToken
+import org.axonframework.eventhandling.tokenstore.ConfigToken
+import org.axonframework.messaging.responsetypes.InstanceResponseType
+import org.axonframework.messaging.responsetypes.MultipleInstancesResponseType
+import org.axonframework.messaging.responsetypes.OptionalResponseType
+import org.axonframework.messaging.responsetypes.ResponseType
+import kotlin.reflect.KClass
+
+private val trackingTokenSerializer = PolymorphicSerializer(TrackingToken::class).nullable
+
+val AxonSerializersModule = SerializersModule {
+    contextual(ConfigToken::class) { ConfigTokenSerializer }
+    contextual(GapAwareTrackingToken::class) { GapAwareTrackingTokenSerializer }
+    contextual(MultiSourceTrackingToken::class) { MultiSourceTrackingTokenSerializer }
+    contextual(MergedTrackingToken::class) { MergedTrackingTokenSerializer }
+    contextual(ReplayToken::class) { ReplayTokenSerializer }
+    contextual(GlobalSequenceTrackingToken::class) { GlobalSequenceTrackingTokenSerializer }
+    polymorphic(TrackingToken::class) {
+        subclass(ConfigTokenSerializer)
```

Comment:
Does this mean for any other token, for example like with the Mongo or Kafka extension, developers have to provide additional configuration?

Reply:
See the other comment on `ScheduleToken`.

Reply:
Just a thought here: When we build the serializer for avro, and came to this point, we agreed that a custom serializer is only relevant for messaging and event-storage, so we decided to just let the default axon-serializer do the lifting for all internal types like tokens ... Its java code anyway.